app/assets/javascripts/admin/enterprises/directives/producer_autocomplete.js.coffee

```diff
@@ -0,0 +1,30 @@
+angular.module("admin.enterprises").directive "ofnProducerAutocomplete", (Enterprises, $sanitize) ->
+  scope: true
+  link: (scope,element,attrs) ->
+    multiple = scope.$eval attrs.multipleSelection
+    placeholder = attrs.placeholder
+    initialSelection = scope.$eval attrs.ngModel
+    suppliers = scope.suppliers
+
+    setTimeout ->
+      element.select2
+        placeholder: placeholder
+        multiple: multiple
+        initSelection: (element, callback) ->
+          if multiple
+            callback Enterprises.findByIDs(initialSelection)
+          else
+            callback Enterprises.findByID(initialSelection)
+        query: (query) ->
+          query.callback { results: Enterprises.findByTerm(suppliers, query.term) }
+        formatResult: (producer) ->
+          $sanitize(producer.name)
+        formatSelection: (producer) ->
+          producer.name
+
+      #Allows drag and drop
+      if multiple
+        element.select2("container").find("ul.select2-choices").sortable
+          containment: 'parent'
+          start: -> element.select2("onSortStart")
+          update: -> element.select2("onSortEnd")
```

Comment:
This is all copy pasted, can you please extract part of this code to a service and use it in both this ofnProducerAutocomplete and  ofnTaxonAutocomplete?
You can for example create a service like autocompleteSelect2 that returns everything in the setTimeout method here and takes a parameter callback and another one called something like list where you can inject the suppliers from this one.

Reply:
I agree with @luisramos0. Did you already work on it @guidoDutra ?

Reply:
Sorry, just saw after ;)